zebra-chain/src/block/commitment.rs

```diff
@@ -108,7 +108,9 @@ impl Commitment {
 
         match NetworkUpgrade::current(network, height) {
             Genesis | BeforeOverwinter | Overwinter => Ok(PreSaplingReserved(bytes)),
-            Sapling | Blossom => Ok(FinalSaplingRoot(sapling::tree::Root(bytes))),
+            Sapling | Blossom => Ok(FinalSaplingRoot(
+                sapling::tree::Root::try_from(bytes).expect("we should have valid bytes"),
```

Comment:
Question: these bytes come from peers, have we already checked they are valid? Or could this panic?

Reply:
I added this at https://github.com/ZcashFoundation/zebra/pull/3544/commits/07b48b3f4761809b598a38b085818bd576cc597d as adding the error there was the easier thing to do however not sure if it is in the right spot.

Reply:
The error looks good to me, I'd like @conradoplg to double-check the cryptographic data conversion.

Reply:
Looks good, thanks!